### ISSUE TITLE
ERROR node instead of match_expression

### ISSUE THREAD

**Issue by gbprod** [tree-sitter/tree-sitter-php#157]
I'm experimenting issue in Neovim indentation with `match` syntax and I wonder if it comes from `match` tree-sitter parsing.

I think the issue comes from the parsing of an empty match expression by this: 

```php
<?php

match (true) {
};
```
is parsed as :

```
(program
  (php_tag)
  (ERROR
    (parenthesized_expression
      (boolean)))
  (empty_statement))
```

The root error node doesn't allows nvim-treesitter to correctly indent the match body as it's not detected as `match_expression`.
I wonder if it should not be parsed as:

```
(program
  (php_tag)
  (expression_statement
    (match_expression
      (parenthesized_expression
        (boolean))
      (match_block
        (ERROR))))))
```

On another hand, php parser (`php --syntax-check`) doesn't trigger syntax error when giving an empty match, maybe we should allows empty match by using `commaSep` instead of `commaSep1`  [here](https://github.com/tree-sitter/tree-sitter-php/blob/670d1eb6822d8c7ade1c71232e0bef42757b9da7/grammar.js#L766). We will obtain that: 

```
    (program
      (php_tag)
      (expression_statement
        (match_expression
          (parenthesized_expression
            (boolean))
          (match_block))))
```

What do you think ?

**Comment by cfroystad** [comment]
Good catch! Your findings are in [accordance with the official parser](https://github.com/php/php-src/blob/master/Zend/zend_language_parser.y#L706)

Unfortunately, I have little time to dedicate to tree-sitter-php at the moment, but I'll be happy to review PRs... Thanks for your contributions so far!

**Comment by gbprod** [comment]
Ok ! I'll make some PR 😉

**Comment by aryx** [comment]
Why does the official parser accept empty match??
PHP ...

**Comment by cfroystad** [comment]
hm, it appears to be [two reasons](https://github.com/php/php-src/pull/5371/files#r407273786) for this strange behavior: "the most consistent with switch" and "this is fine for now".

This brings up the age old question of syntactically vs semantically correct. I'm leaning towards following the official parser and leave the analysis of this always being an error to the next layer. Opinions?

**Comment by gbprod** [comment]
IMHO, if PHP syntax linter does not trigger, error, we should have the same behaviour.

**Comment by aryx** [comment]
I agree. The philosophy of tree-sitter is also usually to balance towards being more tolerant.
